src/lexer/tiger.lex

```diff
@@ -1,7 +1,10 @@
 type pos = int
 type lexresult = Token.token
 
-fun eof() = Token.EOF
+fun eof() =
+    (if SrcString.isBuildingString() then raise SrcString.StringNotClosed(Newline.getPos(SrcString.getStartPos()), Newline.getLine(SrcString.getStartPos()))
+    else if not (SrcComment.closed()) then raise SrcComment.CommentNotClosed(Newline.getPos(hd(SrcComment.getComments())), Newline.getLine(hd(SrcComment.getComments())))
```

Comment:
Maybe, if that's the case then a new `panic` function on `ErrorMsg` might be in order, but error messages are who is responsible for showing issues to the user, spesifically doing the job of filename, and line number look up.